### ISSUE TITLE
Feature request: Launch configuration option to force Java execution environment used for run/debug

### ISSUE THREAD

**Issue by ustieber** [microsoft/vscode-java-debug#884]
Feature request: Please add a way to force a specific Java execution environment (from the environments defined by "java.configuration.runtimes") to be used by the debugger to launch the application.

For compatibility reasons, the Maven project files defines "1.8" as "&lt;java-version&gt;". However, in several environments (HiDPI), the use of at least Java 11 is required to run the application correctly. At the moment, there seems to be no way to force the VSCode Java debugger to use Java 11 and overwrite the "&lt;java-version&gt;" setting from the Maven POM file.

##### Environment
- Operating System: Windows 10
- JDK version: Java SE 1.8 and Java SE 11
- Visual Studio Code version: 1.49.1
- Java extension version: 0.10.0
- Java Debugger extension version: 0.28.0

##### Current Result

Visual Studio Code Java debugger is always launching using Java SE 1.8.

##### Expected Result

Can force the Visual Studio Code Java debugger to use Java 11 to run/debug the application.

**Comment by ustieber** [comment]
Of course, I refer to the "maven.compiler.[source|target|release]" Maven properties. "java-version" is our own Maven property to simplify things.

**Comment by testforstephen** [comment]
The debugger is supposed to pick the same runtime as that the language server used to build your project. Allowing the debugger to hack a different JDK is not what we want.

However, you still can leverage the user setting "java.configuration.runtimes" for your purpose. For example, you can configure JDK 11 for JavaSE-1.8 as below.

```json
  "java.configuration.runtimes": [
    {
      "name": "JavaSE-1.8",
      "path": "/path/to/jdk-11",
    }
  ]
```

**Comment by ustieber** [comment]
Well, that leaves out a set of use cases for the Java tooling. The need to run/debug an application using a different Java runtime than using for the build, is real.

The suggestion works, but that's kind of a hack by itself. If the debugger is sticking to the Java runtime used by the language server, fine. But where is the possibility to easily switch the Java runtime used by the language server? Maven profiles can't be utilized, because currently, there seems to be no way to specify the profile to use by the language server (Maven execution options are ignored by the language server).

Very sad. Because it takes away a lot of comfort from the Java tooling as a whole.

**Comment by testforstephen** [comment]
If you want more flexibility on the launching command line, you could have a try at `preLaunchTask`. See https://github.com/microsoft/vscode-java-debug/issues/120#issuecomment-417269773. With that, you could custom the launching command for your program, and attach the debugger to it.
